src/kagglehub/clients.py

```diff
@@ -171,9 +174,11 @@ def download_file(self, path: str, out_file: str, resource_handle: Optional[Reso
                         _CHECKSUM_MISMATCH_MSG_TEMPLATE.format(expected_md5_hash, actual_md5_hash)
                     )
 
-    def _get_http_basic_auth(self) -> Optional[HTTPBasicAuth]:
+    def _get_auth(self) -> Optional[requests.auth.AuthBase]:
```

Comment:
I like to refactor to rely on AuthBase and have KaggleTokenAuth extend it. Very clean 👏